src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -1065,12 +1064,20 @@ internal static bool IsTrue(IList objectArray)
         }
 
         /// <summary>
-        /// Internal routine that determines if an object meets any of our criteria for null.
+        /// Internal routine that determines if an object meets any of our criteria for true null.
         /// </summary>
         /// <param name="obj">The object to test.</param>
         /// <returns>True if the object is null.</returns>
         internal static bool IsNull(object obj) => obj == null || obj == AutomationNull.Value;
 
+        /// <summary>
+        /// Internal routine that determines if an object meets any of our criteria for null.
+        /// This method additionally checks for <see cref="NullString.Value"/> and <see cref="DBNull.Value"/>
+        /// </summary>
+        /// <param name="obj">The object to test.</param>
+        /// <returns>True if the object is null.</returns>
+        internal static bool IsNullLike(object obj) => obj == DBNull.Value || obj == NullString.Value || IsNull(obj);
```

Comment:
What are the chances of this being public? The `LanguagePrimitives` methods are great for imitating PowerShell logic from C#.

If this was public we (external binary module writers) could:

1. Avoid doing all of those checks ourselves
1. Stop referencing the pseudo internal `AutomationNull`
1. Automatically handle new "null-like"'s when/if more are added

@daxian-dbw any objections?